server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -123,7 +125,8 @@ abstract class AbstractSearchAsyncAction<Result extends SearchPhaseResult> exten
         SearchTask task,
         SearchPhaseResults<Result> resultConsumer,
         int maxConcurrentRequestsPerNode,
-        SearchResponse.Clusters clusters
+        SearchResponse.Clusters clusters,
+        Client client
```

Comment:
Why are we adding this to `AbstractSearchAsyncAction` when only query-then-fetch & DFS need it?

Reply:
Yeap, that's fair; updated in 2eeb251ff9a8c6bce20212ad12ab2409a492f64e.